src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetDateCommand.cs

```diff
@@ -440,11 +440,29 @@ private string UFormatDateString(DateTime dateTime)
                             break;
 
                         case 'G':
-                            sb.Append("{0:yyyy}");
-                            break;
-
                         case 'g':
-                            sb.Append("{0:yy}");
+                            var isoWeek = ISOWeek.GetWeekOfYear(dateTime);
+                            var isoWeekYear = dateTime.Year;
+
```

Comment:
Given the way this logic is used, and the fact that we must finally test again for `'g'` vs `'G'`, I'd suggest restructuring the code as follows:

```csharp
private string UFormatDateString(DateTime dateTime)
{
    ...
    
    case 'G':
        sb.Append(StringUtil.Format("{0:0000}", GetIsoYear(dateTime)));
        break;

    case 'g':
        int isoYearWithoutCentury = GetIsoYear(dateTime) % 100;
        sb.Append(StringUtil.Format("{0:00}", isoYearWithoutCentury);
        break;
    ...
}


/// <summary>
/// Comment describing what problem this method solves
/// </summary>
private static int GetIsoYear(DateTime dateTime)
{
    int isoWeek = ISOWeek.GetWeekOfYear(dateTime);

    // Comment describing why we must correct for week 53
    if (isoWeek == 53 && dateTime.Month == 1)
    {
        return dateTime.Year - 1;
    }

    // Comment describing why week 1/month 12 needs correcting
    if (isoWeek == 1 && dateTime.Month == 12)
    {
        return dateTime.Year + 1;
    }

    return dateTime.Year;
}
```

Reply:
In particular we need to document why we're doing our own datetime handling rather than letting .NET do it